app/cdap/components/WrangleHome/Components/OngoingDataExplorations/utils.ts

```diff
@@ -14,22 +14,34 @@
  * the License.
  */
 
-import { ImportDatasetIcon } from '../WrangleCard/iconStore/ImportDatasetIcon';
-import { IMassagedObject } from './types';
+import DataPrepStore from 'components/DataPrep/store';
+import { IEachData, IMassagedObject } from './types';
 
-export const generateDataForExplorationCard = (oldData) => {
+export const generateDataForExplorationCard = (oldData: IEachData[]) => {
   // Massaging the data to map the API response to the Ongoing Data Exploration List
   const massagedArray = [];
 
+  const { dataprep } = DataPrepStore.getState();
+  const { connectorsWithIcons } = dataprep;
+
+  const getIcon = (connectorName) => {
+    const tempItem = connectorsWithIcons.find((e) => {
+      return e.name === connectorName;
+    });
+    return tempItem.SVG;
+  };
+
   if (oldData && Array.isArray(oldData) && oldData.length) {
     oldData.forEach((eachItem) => {
       const childArray = [];
 
       Object.keys(eachItem).map((keys) => {
         const obj = {} as IMassagedObject;
-
-        if (keys === 'connectionName') {
-          obj.icon = ImportDatasetIcon;
+        if (keys === 'connectorType') {
+          obj.icon = getIcon(eachItem[keys]);
+          obj.label = eachItem[keys];
+          obj.type = 'icon';
+        } else if (keys === 'connectionName') {
```

Comment:
put the strings in constant file and import for checking in conditions

Reply:
Done